src/dev-spaces-preview/azext_dev_spaces_preview/custom.py

```diff
@@ -0,0 +1,154 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from __future__ import print_function
+import os
+import platform
+import subprocess
+import tempfile
+from knack.log import get_logger
+from knack.util import CLIError
+
+logger = get_logger(__name__)
+
+
+# pylint:disable=no-member,too-many-lines,too-many-locals,too-many-statements
+
+
+def aks_use_dev_spaces(cluster_name, resource_group_name, space_name='default', parent_space_name=None):  # pylint: disable=line-too-long
+    """
+    Use Azure Dev Spaces with a managed Kubernetes cluster.
+
+    :param cluster_name: Name of the managed cluster.
+    :type cluster_name: String
+    :param resource_group_name: Name of resource group. You can configure the default group. \
+    Using `az configure –defaults group=<name>`.
+    :type resource_group_name: String
+    :param space_name: Name of the dev space to use.
+    :type space_name: String
+    :param parent_space_name: Name of a parent dev space to inherit from when creating a new dev space. \
+    By default, if there is already a single dev space with no parent, the new space inherits from this one.
+    :type parent_space_name: String
+    """
+
+    azds_tool = 'Azure Dev Spaces CLI (Preview)'
+    should_install_vsce = False
+    system = platform.system()
+    if system == 'Windows':
+        # Windows
+        # Dev Connect Install Path (WinX)
+        azds_cli = os.path.join(os.environ["ProgramW6432"],
```

Comment:
In order to install, does the user need to be admin? What if they run az not as admin? Are they forced to elevate? That's a little weird.

Reply:
We really should be installing a private copy of the CLI inside where our az extension is installed, which is under the user's ~/.azure/cliextensions/dev-spaces-preview directory.

Reply:
Perhaps we have to defer this until all the remaining commands are in the CLI.

Reply:
Currently, we are going with what our usual setup scripts go and using the same location. We can defer it for later and update it then.